src/screens/HomeScreens/components/InfoTable/InfoTable.js

```diff
@@ -1,78 +1,157 @@
+            }
+        }
+
+        
+    }, [checkNum, infos])
 
     return (
         <Container>
-        <h3>Todo List</h3>
-        <br />
-        <div>
-        <Button variant='secondary' onClick={deleteSelectedInfoHandler}>Delete seleted</Button>
-        </div>
-        <br />
-        <Table className="table table-hover">
+            <h3>Todo List</h3>
+            <br />
+            {(infos.length!=0) && 
+            <div>
+            <Row>
+                <Col md={4}>
+                    <Button variant='secondary' disabled={(checkNum===0)? true : false} onClick={deleteSelectedInfoHandler}>Delete seleted</Button> 
+                </Col>
+                <Col md={{ span: 4, offset: 4 }}>
+                    <label >Filter by category: &nbsp;</label>
+                    <select name="category" onChange={filterHandler}> 
+                        <option value="">All</option>
+                        <option value="html">html</option>
+                        <option value="css">css</option>
+                    </select>
+                </Col>
+            </Row>
+            <br />
+            
+            <Table className="table table-hover" >
             <thead>
                 <tr>
                     <th><input type="checkbox" checked={allChecked} onChange={selectAllHandler}/></th>
                     <th>Description</th>
-                    <th>Content</th>
+                    <th>Category</th>
+                    <th>
+                        <button type="button" style={{border: 'none', background:'none'}}  onClick={sortHandler}>
+                            <strong>Deadline</strong>
+                        </button>
+                    </th>                 
                     <th>Operate</th>
                 </tr>
             </thead>
             <tbody>
-                {infos && infos.map(item=>(
-                        <tr key={item.id} className={item.isChecked ? 'selected': ''} >
+                {allinfos.map(item=>(        
+                        <tr className={item.isChecked ? 'selected': ''}  key={item.id} >
                             <td>
                                 <input type="checkbox" value={item.id} onChange={checkedItemHandler} checked={item.isChecked}/>
                             </td>
-                            <LinkContainer to={`/todo/${item.id}`}>
+                            <LinkContainer to={`/todo/${item.id}`}>   
                             <td>{item.description}</td>
-                            </LinkContainer>
+                            </LinkContainer>  
                             <td>{item.category}</td>
+                            
+                            <td>{item.deadline.toDateString()}</td>
                            <td><Button variant="light" onClick={() => deleteInfoHandler(item.id)}>Delete</Button></td>
-                        </tr>
-                
+                        </tr>                          
```

Comment:
can this `<tr>` be extracted into a separate component?

Reply:
In this session ,there are 2 event (onChange and onClick). how to use these events in one Object tag? like<Infolist onChange={} onClick={} />?

Reply:
done